RubeesEat/Model/DB/DbBillRepository.cs

```diff
@@ -0,0 +1,254 @@
+using System.Collections.Immutable;
+using System.Data;
+using System.Data.Common;
+using Microsoft.Data.SqlClient;
+
+namespace RubeesEat.Model.DB;
+
+public class DbBillRepository(IDbConnectionFactory connectionFactory) : IBillRepository
+{
+    public IReadOnlyList<Bill> GetAll()
+    {
+        using var connection = connectionFactory.CreateDbConnection();
+        using var command = connection.CreateCommand();
+
+        command.CommandText = $"""
+                                 SELECT
+                                       B.BillID,
+                                       B.Date,
+                                       B.Description,
+                                       E.EntryLineID,
+                                       E.Amount,
+                                       P.PersonID,
+                                       P.FirstName,
+                                       P.LastName
+                                   FROM
+                                       Bills B
+                                   INNER JOIN
+                                       EntryLines E ON B.BillID = E.BillID
+                                    INNER JOIN
+                                       Persons P ON P.PersonID = E.PersonID
+                                   ORDER BY B.Date DESC, B.BillID;
+                               """;
+        command.CommandType = CommandType.Text;
+        connection.Open();
+        using var reader = command.ExecuteReader();
+        return CreateBillsFromReader(reader);
+    }
+
+    public void Add(Bill bill)
+    {
+        using (var connection = connectionFactory.CreateDbConnection())
+        {
+            connection.Open();
+            using (var command = connection.CreateCommand())
+            {
+                SqlTransaction transaction;
+                
+                transaction = (SqlTransaction)connection.BeginTransaction();
+                
+                command.Connection = connection;
+                command.Transaction = transaction;
```

Comment:
All the transaction setup should probably also be moved into the extension methods.

Reply:
I am a bit confused whether to put this in DbCommand or DbConnection extensions. Also, if I put it in DbConnection extensions, whether or not the two lines 

command.Connection = connection;
command.Transaction = transaction;

should be in the function. I'm worried if it does too much then.

Reply:
It should be easy to use. If we are in a transaction we need to assign the transaction object to the command, so ideally this would be done by the `CreateCommand` method. I don't know if it is a good idea but we could also create a `CreateCommand` method on `DbTransaction`, which could assign the transaction directly. Needing to do it after creating the command is a bit error prone and we should avoid that.

Reply:
We still need the transaction object at the end to do "transaction.Commit" at the end. But if I set the transaction up in CreateCommand, I only receive the command object as return, and not the transaction object. 

public static DbCommand CreateCommand(this DbConnection connection, string sql, IsolationLevel isolationLevel)
    {
        var transaction = (SqlTransaction)connection.BeginTransaction(isolationLevel);
        var command = connection.CreateCommand();
        command.CommandText = sql;
        command.CommandType = CommandType.Text;
        command.Connection = connection;
        command.Transaction = transaction;
        return command;
    }